scripts/terminalserver/terminalserver.jl

```diff
@@ -418,6 +447,52 @@ end
 
 push!(Base.package_callbacks, pkgload)
 
+function hook_repl(repl)
+    main_mode = get_main_mode()
+
+    main_mode.on_done = REPL.respond(Base.active_repl, main_mode; pass_empty = false) do line
+
+        x = Base.parse_input_line(line,filename=REPL.repl_filename(repl,main_mode.hist))
+
+        if !(x isa Expr && x.head == :toplevel)
+            error("VS Code Julia REPL got an unexpected input.")
+        end
+
+        # Replace all top level assignments with a global top level assignment
+        # so that they happen, even though the code now runs inside a
+        # try ... finally block
```

Comment:
FWIW, Juno doesn't bother with all of this and instead does something like
```
  main_mode.on_done = REPL.respond(Base.active_repl, main_mode; pass_empty = false) do line
    if !isempty(line)
      quote
        $evalrepl($mod, $line)
      end
    end
  end
```
```
function evalrepl(mod, line)
  try
    errored = false
    JSONRPC.send_notification(conn_endpoint, "repl/starteval", nothing)
    with_logger(JunoProgressLogger()) do
      try
         line = Meta.parse(line)
      catch err
        errored = true
        display_parse_error(stderr, err)
      end
      errored && return nothing
      try
        repleval(mod, line)
      catch err
        display_error(stderr, err, stacktrace(catch_backtrace()))
      end
    end
    errored ? nothing : ans
  catch err
    # This is for internal errors only.
    display_error(stderr, err, stacktrace(catch_backtrace()))
  finally
    JSONRPC.send_notification(conn_endpoint, "repl/finisheval", nothing)
  end
end
```

I'm not sure which approach I like better though.

Reply:
I'm also unsure :) Should we merge as is for now, and then we can revisit later?

Reply:
Sure.